compilation fails due to Eigen on Debian stable

I have problems compiling opm-autodiff on Debian wheezy with Eigen 3.1.0.
This problem occurs in the current master as well as in @rolk 's buildsystem rollup branch.

```
[  8%] Building CXX object CMakeFiles/opmautodiff.dir/opm/autodiff/FullyImplicitBlackoilSolver.cpp.o
In file included from /usr/include/eigen3/Eigen/src/Core/ArrayBase.h:127:0,
                 from /usr/include/eigen3/Eigen/Core:355,
                 from /usr/include/eigen3/Eigen/Dense:1,
                 from /usr/include/eigen3/Eigen/Eigen:1,
                 from /home/mblatt/tester/opm-autodiff/opm/autodiff/AutoDiffBlock.hpp:24,
                 from /home/mblatt/tester/opm-autodiff/opm/autodiff/FullyImplicitBlackoilSolver.hpp:23,
                 from /home/mblatt/tester/opm-autodiff/opm/autodiff/FullyImplicitBlackoilSolver.cpp:20:
/usr/include/eigen3/Eigen/src/Core/../plugins/ArrayCwiseBinaryOps.h: In instantiation of ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_min_op<typename Eigen::internal::traits<T>::Scalar>, const Derived, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<typename Eigen::internal::traits<T>::Scalar>, Derived> > Eigen::ArrayBase<Derived>::min(const Scalar&) const [with Derived = Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> >; typename Eigen::internal::traits<T>::Scalar = double; Eigen::ArrayBase<Derived>::Scalar = double]’:
/home/mblatt/tester/opm-autodiff/opm/autodiff/FullyImplicitBlackoilSolver.cpp:850:66:   required from here
/usr/include/eigen3/Eigen/src/Core/../plugins/ArrayCwiseBinaryOps.h:39:69: error: could not convert ‘Eigen::ArrayBase<Derived>::min(const Eigen::ArrayBase<OtherDerived>&) const [with OtherDerived = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Array<double, -1, 1> >; Derived = Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> >; typename Eigen::internal::traits<T>::Scalar = double]((*(const Eigen::ArrayBase<Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Array<double, -1, 1> > >*)(& Eigen::DenseBase<Derived>::Constant(Eigen::DenseBase<Derived>::Index, Eigen::DenseBase<Derived>::Index, const Scalar&) [with Derived = Eigen::Array<double, -1, 1>; Eigen::DenseBase<Derived>::ConstantReturnType = Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Array<double, -1, 1> >; typename Eigen::internal::traits<T>::Scalar = double; Eigen::DenseBase<Derived>::Index = long int; Eigen::DenseBase<Derived>::Scalar = double](((const Eigen::EigenBase<Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> > >*)this)->Eigen::EigenBase<Derived>::cols<Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> > >(), (* & other)))))’ from ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_min_op<double>, const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> >, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::Array<double, -1, 1> > >’ to ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_min_op<double>, const Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> >, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> > > >’
/usr/include/eigen3/Eigen/src/Core/../plugins/ArrayCwiseBinaryOps.h: In member function ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_min_op<typename Eigen::internal::traits<T>::Scalar>, const Derived, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<typename Eigen::internal::traits<T>::Scalar>, Derived> > Eigen::ArrayBase<Derived>::min(const Scalar&) const [with Derived = Eigen::CwiseUnaryOp<Eigen::internal::scalar_abs_op<double>, const Eigen::Array<double, -1, 1> >; typename Eigen::internal::traits<T>::Scalar = double; Eigen::ArrayBase<Derived>::Scalar = double]’:
/usr/include/eigen3/Eigen/src/Core/../plugins/ArrayCwiseBinaryOps.h:40:1: warning: control reaches end of non-void function [-Wreturn-type]
make[2]: *** [CMakeFiles/opmautodiff.dir/opm/autodiff/FullyImplicitBlackoilSolver.cpp.o] Fehler 1
make[1]: *** [CMakeFiles/opmautodiff.dir/all] Fehler 2
make: *** [all] Fehler 2
```

--- a/cmake/Modules/opm-autodiff-prereqs.cmake
+++ b/cmake/Modules/opm-autodiff-prereqs.cmake
@@ -19,5 +19,5 @@ set (opm-autodiff_DEPS
 	dune-istl REQUIRED;
 	opm-core REQUIRED"
 	# Eigen
-	"Eigen3 3.1 REQUIRED"
+	"Eigen3 3.1.2 "
 	)

